appveyor.yml

```diff
@@ -28,23 +28,26 @@ before_build:
         }
     - ps: $env:VSCOMNTOOLS=(Get-Content ("env:VS" + "$env:VSVER" + "0COMNTOOLS"))
     - call "%VSCOMNTOOLS%\..\..\VC\vcvarsall.bat" %VCVARS_PLATFORM%
-    - mkdir _build
-    - cd _build
-    - perl ..\Configure %TARGET% %SHARED%
-    - cd ..
     - ps: >-
         if (-not $env:APPVEYOR_PULL_REQUEST_NUMBER`
             -or (&git log -2 | Select-String "\[extended tests\]") ) {
             $env:EXTENDED_TESTS="yes"
+            if ($env:Configuration -NotMatch "shared") {
+                exit 0
+            }
         }
+    - mkdir _build
+    - cd _build
+    - perl ..\Configure %TARGET% %SHARED%
+    - cd ..
 
 build_script:
+    - cd _build
     - ps: >-
         If ($env:Configuration -Match "shared" -or $env:EXTENDED_TESTS) {
-            cd _build
             cmd /c "nmake 2>&1"
-            cd ..
         }
+    - cd ..
```

Comment:
> It's a bit more nuanced there...

I mean that it's not clear that it would be sufficient, because there are multiple commands in one of `if{}` cases...